src/auth/PrivateRoute.jsx

```diff
@@ -0,0 +1,14 @@
+import React, { useContext } from "react";
+import { Route, Redirect } from "react-router-dom";
+import { AuthContext } from "./Authentication";
+
+const PrivateRoute = ({ children, ...rest }) => {
+  const { currentUser } = useContext(AuthContext);
+  return (
+    <Route {...rest}>
+      {!!currentUser ? children : <Redirect to="/signin" />}
```

Comment:
please don't ever use!!
use Boolean

Reply:
esLint did not link Boolean :/ I left it `currentUser ? ....`